Upgrade from 1.7.6.9 to 1.7.8.0 fails according to CI ; Uncaught Error: Call to a member function query() on null

Thanks to @atomiix the upgrade now runs every night against latest nightly build.

According to the dashboard, upgrade from 1.7.6.9 to 1.7.8.0 fails https://nightly.prestashop.com/report/894

Logs https://github.com/PrestaShop/autoupgrade/actions/runs/573249428

```
INFO - === Step upgradeDb
WARNING - 
            <div class="upgradeDbError">
Warning:    [WARNING] SQL 1.7.8.0
            1064 in INSERT IGNORE INTO `ps_hook` (`id_hook`, `name`, `title`, `description`, `position`) VALUES
  (NULL, 'actionPresentCart', 'Cart Presenter', 'This hook is called before a cart is presented', '1'),
  (NULL, 'actionPresentOrder', 'Order footer', 'This hook is called before an order is presented', '1'),
  (NULL, 'actionPresentOrderReturn', 'Order Return Presenter', 'This hook is called before an order return is presented', '1'),
  (NULL, 'actionPresentProduct', 'Product Presenter', 'This hook is called before a product is presented', '1'),
  (NULL, 'displayBanner', 'Display Banner', 'Use this hook for banners on top of every pages', '1'),
  (NULL, 'actionModuleUninstallBefore', 'Module uninstall before', 'This hook is called before module uninstall process', '1'),
  (NULL, 'actionModuleUninstallAfter', 'Module uninstall after', 'This hook is called at the end of module uninstall process', '1'),
  (NULL, 'displayCartModalContent', 'Cart Presenter', 'This hook displays content in the middle of the window that appears a
  ...
WARNING - Warning detected during upgrade.
INFO - Database upgraded. Now upgrading your Addons modules...
INFO - === Step upgradeModules
INFO - 58 modules will be upgraded.
INFO - === Step upgradeModules
INFO - 52 modules left to upgrade.
INFO - === Step upgradeModules
INFO - 45 modules left to upgrade.
INFO - === Step upgradeModules
INFO - 36 modules left to upgrade.
INFO - === Step upgradeModules
INFO - 29 modules left to upgrade.
INFO - === Step upgradeModules
INFO - 22 modules left to upgrade.
INFO - === Step upgradeModules
INFO - 15 modules left to upgrade.
INFO - === Step upgradeModules
INFO - 7 modules left to upgrade.
INFO - === Step upgradeModules
INFO - === Step upgradeModules
INFO - Addons modules files have been upgraded.
INFO - === Step cleanDatabase
INFO - The database has been cleaned.
INFO - === Step upgradeComplete
INFO - Upgrade process done, but some warnings have been found.
PHP Notice:  Undefined property: DbPDO::$link in /var/www/html/classes/db/DbPDO.php on line 165
PHP Fatal error:  Uncaught Error: Call to a member function query() on null in /var/www/html/classes/db/DbPDO.php:165
Stack trace:
#0 /var/www/html/classes/db/Db.php(376): DbPDOCore->_query('SELECT c.`name`...')
#1 /var/www/html/classes/db/Db.php(613): DbCore->query('SELECT c.`name`...')
#2 /var/www/html/classes/Configuration.php(160): DbCore->executeS('SELECT c.`name`...')
#3 /var/www/html/classes/Configuration.php(209): ConfigurationCore::loadConfiguration()
#4 /var/www/html/classes/Cookie.php(389): ConfigurationCore::get('PS_COOKIE_SAMES...')
#5 /var/www/html/classes/Cookie.php(455): CookieCore->encryptAndSetCookie('date_add|2021-0...')
#6 /var/www/html/classes/Cookie.php(423): CookieCore->write()
#7 [internal function]: CookieCore->__destruct()
#8 ***main***
  thrown in /var/www/html/classes/db/DbPDO.php on line 165
Error: Process completed with exit code 255.
```

--- a/classes/Cookie.php
+++ b/classes/Cookie.php
@@ -54,6 +54,9 @@ class CookieCore
     /** @var array Website domain for setcookie() */
     protected $_domain;
 
+    /** @var string|bool SameSite for setcookie() */
+    protected $_sameSite;
+
     /** @var array Path for setcookie() */
     protected $_path;
 
@@ -89,6 +92,7 @@ class CookieCore
         $this->_path = rawurlencode($this->_path);
         $this->_path = str_replace(['%2F', '%7E', '%2B', '%26'], ['/', '~', '+', '&'], $this->_path);
         $this->_domain = $this->getDomain($shared_urls);
+        $this->_sameSite = Configuration::get('PS_COOKIE_SAMESITE');
         $this->_name = 'PrestaShop-' . md5(($this->_standalone ? '' : _PS_VERSION_) . $name . $this->_domain);
         $this->_allow_writing = true;
         $this->_salt = $this->_standalone ? str_pad('', 32, md5('ps' . __FILE__)) : _COOKIE_IV_;
@@ -386,8 +390,6 @@ class CookieCore
             $time = 1;
         }
 
-        $sameSite = Configuration::get('PS_COOKIE_SAMESITE');
-
         /*
          * The alternative signature supporting an options array is only available since
          * PHP 7.3.0, before there is no support for SameSite attribute.
@@ -398,7 +400,7 @@ class CookieCore
                 $content,
                 $time,
                 $this->_path,
-                $this->_domain . '; SameSite=' . $sameSite,
+                $this->_domain . '; SameSite=' . $this->_sameSite,
                 $this->_secure,
                 true
             );
@@ -413,7 +415,7 @@ class CookieCore
                 'domain' => $this->_domain,
                 'secure' => $this->_secure,
                 'httponly' => true,
-                'samesite' => $sameSite,
+                'samesite' => $this->_sameSite,
             ]
         );
     }

--- a/install-dev/upgrade/sql/1.7.8.0.sql
+++ b/install-dev/upgrade/sql/1.7.8.0.sql
@@ -20,7 +20,7 @@ INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `po
   (NULL, 'actionOverrideEmployeeImage', 'Override Employee Image', 'This hook is used to override the employee image', '1'),
   (NULL, 'actionFrontControllerSetVariables', 'Add variables in JavaScript object and Smarty templates', 'Add variables to javascript object that is available in Front Office. These are also available in smarty templates in modules.your_module_name.', '1'),
   (NULL, 'displayAdminGridTableBefore', 'Display before Grid table', 'This hook adds new blocks before Grid component table.', '1'),
-  (NULL, 'displayAdminGridTableAfter', 'Display after Grid table', 'This hook adds new blocks after Grid component table.', '1')
+  (NULL, 'displayAdminGridTableAfter', 'Display after Grid table', 'This hook adds new blocks after Grid component table.', '1'),
   (NULL, 'displayAdminOrderCreateExtraButtons', 'Add buttons on the create order page dropdown', 'Add buttons on the create order page dropdown', '1')
 ;
 

